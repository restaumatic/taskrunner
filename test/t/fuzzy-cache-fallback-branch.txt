# no toplevel
# s3

# Test similar to `fuzzy-cache`, but the second run is done on a different branch.
# We still want fuzzy cache from the main branch.

export TASKRUNNER_SAVE_REMOTE_CACHE=1

mkdir a

(
  cd a
  echo foo > input.txt
  git init -q -b main
  git add input.txt
)

cp -r a b

(
  cd b
  git checkout -q -b feature-branch
)

go() {
  export TASKRUNNER_FALLBACK_BRANCHES="main"
  # Note: we want separate workdir for separate repos
  TASKRUNNER_STATE_DIRECTORY="$(pwd)" taskrunner -n mytask bash -e -c '
    snapshot input.txt --outputs output.txt --fuzzy-cache 
    echo -n "Expensive computation, "
    if [ -f output.txt ]; then
      echo "previous output is $(cat output.txt)"
    else
      echo "no previous output"
    fi
    rev input.txt > output.txt
  '
}

(
  cd a
  go
  echo "Output in a is: $(cat output.txt)"
)

(
  cd b
  echo bar > input.txt
  go # note: should run "expensive computation" again, but having previous output.txt available
  echo "Output in b is: $(cat output.txt)"
)
