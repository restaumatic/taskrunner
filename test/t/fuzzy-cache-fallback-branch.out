-- output:
[mytask] stdout | Expensive computation, no previous output
Output in a is: oof
[mytask] stdout | Expensive computation, previous output is oof
Output in b is: rab
